internal/cleanup/cleanup.go

```diff
@@ -133,49 +132,75 @@ type exportCleanupHandler struct {
 }
 
 func (h *exportCleanupHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	ctx, span := trace.StartSpan(r.Context(), "(*cleanup.exportCleanupHandler).ServeHTTP")
-	defer span.End()
+	ctx := r.Context()
+	logger := logging.FromContext(ctx).Named("cleanup.export")
 
-	logger := logging.FromContext(ctx)
-
-	cutoff, err := cutoffDate(ctx, h.config.TTL, h.config.DebugOverrideCleanupMinDuration)
+	cutoff, err := cutoffDate(h.config.TTL, h.config.DebugOverrideCleanupMinDuration)
 	if err != nil {
-		message := fmt.Sprintf("error calculating cutoff time: %v", err)
-		stats.Record(ctx, mExportsSetupFailed.M(1))
-		logger.Error(message)
-		span.SetStatus(trace.Status{Code: trace.StatusCodeInternal, Message: message})
-		http.Error(w, "internal processing error", http.StatusInternalServerError)
+		logger.Errorw("failed to calculate cutoff date", "error", err)
+		respondWithError(w, http.StatusInternalServerError, err)
 		return
 	}
-	logger.Infof("Starting cleanup for export files older than %v", cutoff.UTC())
-	stats.Record(ctx, mExportsCleanupBefore.M(cutoff.Unix()))
 
-	// Set h.Timeout
-	timeoutCtx, cancel := context.WithTimeout(ctx, h.config.Timeout)
-	defer cancel()
+	// Construct a multi-error. If one of the purges fails, we still want to
+	// attempt the other purges.
+	var merr *multierror.Error
 
-	count, err := h.database.DeleteFilesBefore(timeoutCtx, cutoff, h.blobstore)
-	if err != nil {
-		message := fmt.Sprintf("Failed deleting export files: %v", err)
-		logger.Error(message)
-		stats.Record(ctx, mExposuresDeleteFailed.M(1))
-		span.SetStatus(trace.Status{Code: trace.StatusCodeInternal, Message: message})
-		http.Error(w, "internal processing error", http.StatusInternalServerError)
+	// Files
+	func() {
+		ctx, cancel := context.WithTimeout(ctx, h.config.Timeout)
+		defer cancel()
+
+		if count, err := h.database.DeleteFilesBefore(ctx, cutoff, h.blobstore); err != nil {
+			merr = multierror.Append(merr, fmt.Errorf("failed to delete files: %w", err))
+		} else {
+			logger.Infow("purged files", "count", count)
+		}
+	}()
+
+	if merr != nil {
+		logger.Errorw("failed to cleanup exposures", "errors", merr.WrappedErrors())
```

Comment:
this is export files, not exposures, right?

Reply:
Yea